### ISSUE TITLE
Command suggestions for native command testing

### ISSUE THREAD

**Issue by theJasonHelmick** [PowerShell/Crescendo#10]
The Native Command Proxy framework is designed to reduce the maintenance cost and burden of wrapping native commands for a PowerShell-like cmdlet experience.

From customer feedback, we’ve identified a series of high value linux commands related to the installation of packages.  In an effort to test the native command framework, we should consider testing against one or several of these commands.

Commands:
- dpkg
- dpkg-query
- rpm
- apt
- dnf
- zypper
- snap

@JamesWTruher , while we cannot work with all of these in our initial testing, what do you think of focusing on apt, specifically apt install?

**Comment by aaronsb** [comment]
I'd like to add pkcon to the list of package managers (the one that KDE relies upon). You'll notice it already has a verb-noun structure in it's options.

**Comment by awakecoding** [comment]
Here are a couple of commands:

**"docker"** which I made my own wrapper for my needs, using PowerShell classes inspired by the docker compose format. I then pass those objects into helper functions that generate my final "docker run" commands and it works very well, with a nice PowerShell look and feel: https://github.com/Devolutions/WaykBastion-ps/blob/master/WaykBastion/Private/DockerHelper.ps1

Commands like docker have advanced JSON functionality built-in, so most administrators using bash will call 'jq' to process the output. PowerShell has ConvertFrom-Json, so 'jq' is not needed.

**"netstat"** which is native to all platforms, with differences in parameter names and output formatting. I made a wrapper and ported it to Windows/macOS/Linux: https://gist.github.com/awakecoding/14ae283e0018a6e9e77ba23c1f4d26f2

Whatever Crescendo becomes, it should definitely make it possible to handle a lot of platform-specific differences, or parameters that may not be present on specific versions of the CLI. There's no way around using if/else switches on those, but if there are ways to make it easier, I wouldn't mind.

**"bcdedit"** unbeliveable but true, this command that lets you edit boot entries (and therefore create safe boot entries) has no PowerShell equivalent, and to make it truly difficult, it has a localized CLI interface. Try parsing the output on systems in French or German, that doesn't work well at all. It is one of the first CLIs I wrapped in PowerShell, and I would definitely rewrite all of that code by now to use the call operator instead: https://github.com/Devolutions/WaykAgent-ps/blob/fdca9595797b2318ff5904b9ec6f24f43d69174a/WaykNow/Private/BcdEdit.ps1

As a workaround for the fact that bcdedit.exe is localized, I did something ugly that could probably be natively supported: I intentionally broke the MUI by copying bcdedit.exe inside a temporary folder where the corresponding non-English MUI file could not be loaded, forcing the output to English. There is probably a way to load the executable in such a way that we could force the language, making it easier to wrap localized CLIs.
